### ISSUE TITLE
ulimit and stack size

### ISSUE THREAD

**Issue by askhamwhat** [microsoft/WSL#633]
When I initiate a bash session, the output of 

```
ulimit -s
```

is 8192 (which is in kb). This limit cannot be increased (Ideally I'd like to be able to set it as unlimited). Any suggestions?

**Comment by sunilmut** [comment]
@askhamwhat - The stack limit on WSL is currently static. Please provide this feedback on our [user voice] (https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo) page as well to help us prioritize this going forward.

**Comment by AndreiLux** [comment]
I'm currently this limitation in a hard way - there seems to be no way to run WRF on WSL with the stack limit currently set this small. I'm setting the stack to higher values at compile time but obviously it affect the behaviour of the program, still segfaulting seconds into a run.

**Comment by eszwabow** [comment]
I found a workaround to this but it requires access to root.  If you can login as root (su -) then u can add yourself to the sudo group if not able to yet (usermod -aG sudo $USER_NAME) then update the stack limit as root (ulimit -s unlimited) and login back into your user (su - $USER_NAME) then you can set the stack limit with: ulimit -s unlimited.

Verify it works with ulimit -a.

I've been using this so that I can run parallel CFD simulations locally on my machine. Good luck!

**Comment by Trass3r** [comment]
`sudo bash` is an easier way to run the ulimit command. `ulimit -a` does show the correct value but haven't tested it in practice yet.

**Comment by Trass3r** [comment]
> @askhamwhat - The stack limit on WSL is currently static. Please provide this feedback on our [user voice](https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo) page as well to help us prioritize this going forward.

For the record: https://wpdev.uservoice.com/forums/266908-command-prompt-console-windows-subsystem-for-l/suggestions/15125013-allow-changes-to-stack-size-via-the-ulimit-command

**Comment by cdzhan** [comment]
> > @askhamwhat - The stack limit on WSL is currently static. Please provide this feedback on our [user voice](https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo) page as well to help us prioritize this going forward.
> 
> For the record: https://wpdev.uservoice.com/forums/266908-command-prompt-console-windows-subsystem-for-l/suggestions/15125013-allow-changes-to-stack-size-via-the-ulimit-command

@askhamwhat is this problem solved？

**Comment by p-maybank** [comment]
@benhillis - I would like to be able to change the stack size, similarly to what other WSL users have requested on this thread.  I get the following error message,
```
$ ulimit -s 10000
-bash: ulimit: stack size: cannot modify limit: Invalid argument
```
Do you have any suggestions?

**Comment by therealkenc** [comment]
![image](https://user-images.githubusercontent.com/7049700/84691862-b8e00400-aef9-11ea-88ae-644831bc0db5.png)

And yes, the second call will fail if you try to grow a hard limit after you shrink it. That is the purpose of ulimit.

**Comment by sblisesivdin** [comment]
This problem still exists and can be solved (at least for Ubuntu 20.04) for WSL2 with:

`sudo prlimit --stack=unlimited --pid $$; ulimit -s unlimited`

You need to run this for every session.

**Comment by Lollith** [comment]
> This problem still exists and can be solved (at least for Ubuntu 20.04) for WSL2 with:
> 
> `sudo prlimit --stack=unlimited --pid $$; ulimit -s unlimited`
> 
> You need to run this for every session.

WSL1 : OK too
